torch/csrc/jit/script/compilation_unit.h

```diff
@@ -201,5 +207,20 @@ struct TORCH_API CompilationUnit {
 };
 
 } // namespace script
+
+// An owning pointer to a Function. Just a pair of a raw Function ptr and it's
+// owning CU. We need this because pybind requires a ref-counted way to refer to
+// Functions.
+struct StrongFunctionPtr {
+  StrongFunctionPtr(
+      std::shared_ptr<script::CompilationUnit> cu,
+      Function* function)
+      : cu_(std::move(cu)), function_(function) {
+    TORCH_INTERNAL_ASSERT(cu_);
+    TORCH_INTERNAL_ASSERT(function_);
+  }
+  std::shared_ptr<script::CompilationUnit> cu_;
+  Function* function_;
```

Comment:
Should this class have operator* and operator->() implemented?

Reply:
Since it's not supposed to be a widely used API I think we should avoid obfuscating what it really is behind operator overloads. If it turns out to be too clunky I can add.